sui-execution/latest/sui-move-natives/src/crypto/twopc_mpc.rs

```diff
@@ -200,3 +206,37 @@ pub fn sign_verify_encrypted_signature_parts_prehash(
 
     Ok(NativeResult::ok(cost, smallvec![Value::bool(valid),]))
 }
+
+pub fn verify_signatures_native(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 4);
+    let sign_cost_params = &context
+        .extensions()
+        .get::<NativesCostTable>()
+        .sign_cost_params
+        .clone();
+    native_charge_gas_early_exit!(context, sign_cost_params.verify_signatures_cost_base);
```

Comment:
@omersadika I think that we shouldn't charge anything here, since this is a part of the sign flow and fees are charged there anyways.

Reply:
I don't think we can even charge here before the system should call it

Reply:
AFAIU, the `native_charge_gas_early_exit ` function purpose is to charge the native function gas cost in the beginning, as it does not charge the gas automatically on each step as the move VM does. If I'll remove it, a validator will may run the entire native function code, just to discover in the end the sender didn't have enough gas to cover the costs. So shall I remove it? Am I wrong here? @ycscaly @omersadika

Reply:
But who is the sender here? it is a validator right? it shouldn't pay fees. The one that should pay fees is the user that asked for the sign protocol, and he already did